lib/afinter.c

```diff
@@ -496,3 +501,74 @@ afinter_global_deinit(void)
     }
   current_internal_source = NULL;
 }
+
+/****************************************************************************
+ * Live collection of internal log messages.
+ ****************************************************************************/
+
+guint8
+afinter_start_live_collection(void)
+{
+  AFInterSource *self;
+
+  if (is_live_collection)
+    return AF_LIVE_COLLECTION_RUNNING;
+
+  if (current_internal_source != NULL)
+    return AF_INTERNAL_SRC_PRESENT;
+
+  self = g_new0(AFInterSource, 1);
+
+  g_static_mutex_lock(&internal_msg_lock);
+  current_internal_source = self;
+  g_static_mutex_unlock(&internal_msg_lock);
+
+  is_live_collection = TRUE;
+
+  return AF_LIVE_COLLECTION_STARTED;
+}
+
+void
+afinter_reset_live_collection(void)
+{
+  afinter_stop_live_collection();
+
+  if (internal_msg_queue)
+    _release_internal_msg_queue();
+}
+
+void
+afinter_stop_live_collection(void)
+{
+  if (is_live_collection)
+    {
+      if (current_internal_source != NULL)
+        {
+          AFInterSource *self = current_internal_source;
+
+          g_static_mutex_lock(&internal_msg_lock);
+          current_internal_source = NULL;
+          g_static_mutex_unlock(&internal_msg_lock);
+
+          is_live_collection = FALSE;
+          g_free(self);
+        }
+    }
+}
+
+void
+afinter_get_collected_messages(GString *result)
+{
+  LogMessage *msg;
+  const gchar *msg_text;
+
```

Comment:
What happens if `afinter_message_posted` is called while this function executes?

Reply:
@Kokan : In that case, it could create an issue when ```afinter_message_posted``` tries to push message and this function tried to get those at the same time.
May be, I should use ```g_static_mutex_lock(&internal_msg_lock);``` in ```afinter_get_collected_messages``` function